src/Gambling.cpp

```diff
@@ -138,9 +138,10 @@ void Gambling::WinGamble(Player *_Player) {
     // Executes the events when the gamble is won.
     // Uses the Player object to give player items won.
     cout << "You win the bet!" << endl;
-    vector<int> drops;
+    ItemMap_t drops;
     for (int i = 0; i < NUM_ITEMS; i++) {
-        drops.push_back((Item == i) ? ItemNumber : 0);
+        ITEMTYPE itemType = static_cast<ITEMTYPE>(i + 1);
+        drops[itemType] = (Item == i) ? ItemNumber : 0;
```

Comment:
I see how static_cast is kind of unavoidable here (without changing the `int Item` private variable in `Gambling.h` - it should really be `ITEMTYPE Item`). You have a few options here, you can either change `int Item` -> `ITEMTYPE Item` and the associated code, or keep the `static_cast` here. The 2nd option might be out of scope for this PR, so if you decide to keep the static cast I'll create a new issue to address this.

Since a gamble only returns one drop, I think something like this would be sufficient (no need for the for loop):

```c++
map<ITEMTYPE, int> drops;
ITEMTYPE itemType = static_cast<ITEMTYPE>(i + 1);
drops[itemType] = ItemNumber;
```

Reply:
@Horki the for loop is still here? This should be changed as per my comment above